### DISCUSSION TITLE
Does PS3 Controller supports any kind of Bluetooth encryption?

### DISCUSSION THREAD

**Discussion by lvella** [nefarius/BthPS3#100] [Q&A]
It is my understand that PC drivers for PS3 controller (either this one or Linux's BlueZ) does not use any kind of encryption when talking to the device via bluetooth. Is this correct? Do you know if the PS3 controller supports any kind of encryption (proprietary or not) that could prevent a device from being spoofed?

**Comment by nefarius** [comment]
I have no idea if the actual traffic is encrypted because I do not need to know that; at the point where I am processing the data, the Bluetooth chip and firmware have already taken care about the Layer 1 & 2 challenges, essentially abstracting away from us what really happens on the hypothetical wire. Since to my knowledge there is no pre-shared key or other complex key exchange happening, it might as well go over the air in cleartext.

**Reply by nefarius** [reply]
To clarify; if we're talking about Transport Layer Security/Encryption. The actual payload (the data buffers) is not encrypted/obfuscated in any way, shape or form.
